src/Symfony/Component/Serializer/Tests/Fixtures/FooInterfaceDummyDenormalizer.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Tests\Fixtures;
+
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class FooInterfaceDummyDenormalizer implements DenormalizerInterface
+{
+    public function denormalize(mixed $data, string $type, string $format = null, array $context = []): array
+    {
+        $result = [];
+        foreach ($data as $foo) {
+            $fooDummy = new FooImplementationDummy();
+            $fooDummy->name = $foo['name'];
+            $result[] = $fooDummy;
+        }
+
+        return $result;
+    }
+
+    public function supportsDenormalization(mixed $data, string $type, string $format = null, array $context = []): bool
+    {
+        if (str_ends_with($type, '[]')) {
+            $className = substr($type, 0, -2);
+            $classImplements = class_implements($className);
+            \assert(\is_array($classImplements));
```

Comment:
I am curious. What this assertion is aimed to do?  What is the point of using this here?

Reply:
yep, what @mtarld said :smile:

Reply:
@andersmateusz This was taken out from the bundle that we use which reproduced this issue. I bet that PHPStan (which that bundle uses) was throwing an error here because `class_implements` can return `false` and PHPStan is not smart enough to know that the second condition will not be executed in that case.